pkg/deploy/deploy.go

```diff
@@ -31,11 +31,94 @@ import (
 	"github.com/spf13/afero"
 )
 
-func Deploy(workingDir string, fs afero.Fs, environmentsFile string,
-	specificEnvironment string, proj string, dryRun bool, continueOnError bool) error {
-	environments, errors := environment.LoadEnvironmentList(specificEnvironment, environmentsFile, fs)
+type Deploy struct {
```

Comment:
Honestly I didn't yet get to fully review this file due to the amount of changes you're introducing by turning this from simple functions into an Object, but: I think you could achieve the test-ability without turning this into an instantiated Deploy struct by keeping the function type definitions you might need for mocking/dep inversion and passing them into the methods that need them.

I see no _real_ need for the big structural and conceptual change introduced here - but the other comments in this review are more important that this one.

If I find the time I'll check out your fork and see if I can make a proposal how that would look.